maven-model-builder/src/test/java/org/apache/maven/model/profile/activation/FileProfileActivatorTest.java

```diff
@@ -0,0 +1,142 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests {@link FileProfileActivator}.
+ *
+ * @author Ravil Galeyev
+ */
+public class FileProfileActivatorTest extends AbstractProfileActivatorTest<FileProfileActivator>
+{
+    private static final String FILE = "file.txt";
+
+    @TempDir
+    Path tempDir;
+
+    private final DefaultProfileActivationContext context = new DefaultProfileActivationContext();
+
+    public FileProfileActivatorTest()
+    {
+        super( FileProfileActivator.class );
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception
+    {
+        super.setUp();
+        activator.setProfileActivationFilePathInterpolator(
+                new ProfileActivationFilePathInterpolator().setPathTranslator( new DefaultPathTranslator() ) );
+
+        context.setProjectDirectory( new File( tempDir.toString() ) );
+
+        File file = new File( tempDir.resolve( FILE ).toString() );
+        if ( !file.exists() )
+        {
+            if ( !file.createNewFile() )
+            {
+                throw new IOException( "Can't create " + file );
+            }
+        }
+    }
+
+
+    @Test
+    public void test_isActive_noFile()
+    {
+        assertActivation( false, newExistsProfile( null ), context );
+        assertActivation( false, newExistsProfile( "someFile.txt" ), context );
+        assertActivation( false, newExistsProfile( "${basedir}/someFile.txt" ), context );
+
+        assertActivation( false, newMissingProfile( null ), context );
+        assertActivation( true, newMissingProfile( "someFile.txt" ), context );
+        assertActivation( true, newMissingProfile( "${basedir}/someFile.txt" ), context );
+    }
+
+    @Test
+    public void test_isActiveExists_fileExists()
+    {
+        assertActivation( true, newExistsProfile( FILE ), context );
+        assertActivation( true, newExistsProfile( "${basedir}" ), context );
+        assertActivation( true, newExistsProfile( "${basedir}/" + FILE ), context );
+
+        assertActivation( false, newMissingProfile( FILE ), context );
+        assertActivation( false, newMissingProfile( "${basedir}" ), context );
+        assertActivation( false, newMissingProfile( "${basedir}/" + FILE ), context );
+    }
+
+    @Test
+    public void test_isActiveExists_leavesFileUnchanged()
+    {
+        Profile profile = newExistsProfile( FILE );
+        assertEquals( profile.getActivation().getFile().getExists(), FILE );
+
+        assertActivation( true, profile, context );
+
+        assertEquals( profile.getActivation().getFile().getExists(), FILE );
+    }
+
+    private Profile newExistsProfile( String filePath )
+    {
+        return newProfile( filePath, true );
+    }
+
+    private Profile newMissingProfile( String filePath )
+    {
+        return newProfile( filePath, false );
+    }
+
+    private Profile newProfile( String filePath, boolean exists )
```

Comment:
Instead of an exists argument, consider two methods: newProfile and newMissingProfile or something like that.

Reply:
there are `newMissingProfile` and `newExistsProfile`. They both call `newProfile`. Anyway, I refactored it and removed the boolean parameter.